.pre-commit-config.yaml

```diff
@@ -179,8 +179,12 @@ repos:
             Part of the Carbon Language project, under the Apache License v2.0 with LLVM
             Exceptions. See /LICENSE for license information.
             SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+          - --skip_pattern
+          # The default skip pattern skips JSON which we use with VSCode and
+          # it, like most file-based parsers, can support comments easily.
+          - '(^LICENSE|\.(ico))$'
           - --custom_format
-          - '\.(carbon|proto|ypp)(\.tmpl)?$'
+          - '\.(carbon|c|json|proto|ypp)(\.tmpl)?$'
```

Comment:
FWIW, many consumers of JSON, at least the JSON in our project, specifically use C++ line style comments from what I can see. We already had those in one place. I tried to explain this above, but happy to improve that comment if it wasn't clear?

My intent here was for us to exclude any JSON files that need to follow strict JSON syntax and cannot support comments. I added two such below as exceptions.

Reply:
Wild, I hadn't seen that GH highlighted these as incorrect.

Amazingly, GitHub doesn't seem to complain about the comments here:
https://github.com/carbon-language/carbon-lang/blob/trunk/utils/vscode/language-configuration.json

I have no idea why it complains in one place and not the other.

I'm worried about using the `"_copyright"` because that too is just a convention, and I'm worried that there will be schemas that don't support it. =/

I feel like we should use `//` comments where they work, and just not worry about the file header where they don't work. I guess the question becomes whether we:

- Manually exclude all the `.json` files where they don't work. I'll add `extensions.json` to the list, and we can keep adding more as discovered.
- Manually include the few `.json` files where they *do* work. I can try to figure out how to do this with the precommit stuff.
- Just never use a header in a `.json` file. I'll change this PR to remove the one case we have then.

What do you think? I'm happy with any of these, although I lean towards the first or third as those are the ones I know how to do.

Reply:
We can try (1)

Reply:
Sounds good, should be implemented now.